tools/generators/migrate-converged-pkg/index.ts

```diff
@@ -225,32 +228,73 @@ function normalizeOptions(host: Tree, options: AssertedSchema) {
   };
 }
 
-function validateUserInput(tree: Tree, options: MigrateConvergedPkgGeneratorSchema): asserts options is AssertedSchema {
-  if (options.name && options.stats) {
+/**
+ *
+ * Narrows down Schema definition to true runtime shape after {@link validateSchema} is executed
+ * - also properly checks of truthiness of provided value
+ */
+function hasSchemaFlag<T extends MigrateConvergedPkgGeneratorSchema, K extends keyof T>(
+  schema: T,
+  flag: K,
+): schema is T & Record<K, NonNullable<T[K]>> {
+  return Boolean(schema[flag]);
+}
+
+async function validateSchema(tree: Tree, schema: MigrateConvergedPkgGeneratorSchema) {
+  let newSchema = { ...schema };
+
+  if (newSchema.name && newSchema.stats) {
     throw new Error('--name and --stats are mutually exclusive');
   }
 
-  if (options.name && options.all) {
+  if (newSchema.name && newSchema.all) {
     throw new Error('--name and --all are mutually exclusive');
   }
 
-  if (options.stats && options.all) {
+  if (newSchema.stats && newSchema.all) {
     throw new Error('--stats and --all are mutually exclusive');
   }
 
-  if (!options.name && !(options.all || options.stats)) {
+  const shouldValidateNameInput = () => {
+    return !newSchema.name && !(newSchema.all || newSchema.stats);
+  };
+
+  const shouldTriggerPrompt = arePromptsEnabled() && shouldValidateNameInput();
+
+  // console.error(newSchema, parsedArgs);
+  if (shouldTriggerPrompt) {
+    const schemaPromptsResponse = await triggerDynamicPrompts();
+
+    newSchema = { ...newSchema, ...schemaPromptsResponse };
```

Comment:
not a fan of using `let` with mutation or even changing reference value completely (this case), but it's most concise solution for this use-case